src/net/posix-stack.cc

```diff
@@ -1093,6 +1098,15 @@ std::vector<network_interface> posix_network_stack::network_interfaces() {
     return std::vector<network_interface>(thread_local_interfaces.begin(), thread_local_interfaces.end());
 }
 
+future<> posix_network_stack::initialize() {
+    engine().at_exit([] {
+        return exiting_gate.close().then([] {
+            posix_ap_server_socket_impl::conn_q.clear();
```

Comment:
Why is this `clear()` needed? If we're on main thread, this queue is empty, if on the auxiliary then the check in `move_connected_socket` makes sure no entries are there any longer.

Reply:
@xemul sorry about the delay getting back to you!

the primary purpose of the gate on the main thread is not to fix the race definitively, but to introduce a barrier such that when a socket is closed during shutdown that we can be sure that if the race occurred then the entry is on the other core early enough that we can...

> Why is this clear() needed?

clean up with `conn_q.clear()` before the reactor thread exits and schedules more continuations on a shutdown reactor.

### How could the race be completely fixed?

I worked on this a bit in the realm of more elaborate internal accounting tricks, but they all started to spiral out of control. Ultimately it seems like there might need to be a change at the socket API level such that `abort_accept` synchronizes across cores in some fashion.